### DISCUSSION TITLE
sample curl for calling workflow definitions from postman

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5340] [Q&A]
Share sample curl for calling workflow definitions from postman

**Comment by glime-ai** [comment]
Certainly! Below is a sample `curl` command that you can use in Postman to call a workflow definition. Please replace `{workflow_definition_id}` with the actual ID of the workflow definition you want to execute and `{api_key}` with your actual API key:

```shell
curl --location --request POST 'https://localhost:5001/elsa/api/workflow-definitions/{workflow_definition_id}/execute' \
--header 'Content-Type: application/json' \
--header 'Authorization: ApiKey {api_key}' \
--data-raw '{}'
```

To use this in Postman, you would:

1. Open Postman.
2. Create a new request by clicking the "New" button and selecting "Request".
3. Set the HTTP method to `POST`.
4. Enter the URL `https://localhost:5001/elsa/api/workflow-definitions/{workflow_definition_id}/execute` (replace `{workflow_definition_id}` with the actual ID).
5. Under "Headers", add two key-value pairs:
   - `Content-Type` with the value `application/json`
   - `Authorization` with the value `ApiKey {api_key}` (replace `{api_key}` with your actual API key).
6. In the "Body" section, select "raw" and enter `{}` or the appropriate JSON payload if the workflow requires input.
7. Click "Send" to execute the workflow.

Make sure your Elsa Workflow server is running and accessible at the specified URL when you send the request.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
